src/workerd/api/node/i18n.c++

```diff
@@ -40,100 +40,130 @@ const char* getEncodingName(Encoding input) {
   JSG_REQUIRE(limit <= ISOLATE_LIMIT, Error, "Source buffer is too large to transcode");
-  auto out = kj::heapArray<kj::byte>(limit);
-  char* target = out.asChars().begin();
+
+  auto out = jsg::BackingStore::alloc<v8::Uint8Array>(js, limit);
+  auto outPtr = out.asArrayPtr().asChars();
+  char* target = outPtr.begin();
   const char* source_ = source.asChars().begin();
   UErrorCode status{};
   ucnv_convertEx(to.conv(), from.conv(), &target, target + limit, &source_, source_ + source.size(),
       nullptr, nullptr, nullptr, nullptr, true, true, &status);
   if (U_SUCCESS(status)) {
-    return out.first(target - out.asChars().begin()).attach(kj::mv(out));
+    out.limit(target - outPtr.begin());
+    return jsg::BufferSource(js, kj::mv(out));
   }
 
   return kj::none;
 }
 
-kj::Maybe<kj::Array<kj::byte>> TranscodeLatin1ToUTF16(
-    kj::ArrayPtr<kj::byte> source, Encoding fromEncoding, Encoding toEncoding) {
-  auto length_in_chars = source.size() * sizeof(UChar);
+kj::Maybe<jsg::BufferSource> TranscodeLatin1ToUTF16(
+    jsg::Lock& js, kj::ArrayPtr<kj::byte> source, Encoding fromEncoding, Encoding toEncoding) {
+  auto length_in_chars = source.size() * sizeof(char16_t);
   // Workers are limited to 128MB so this isn't actually a realistic concern, but sanity check.
   JSG_REQUIRE(length_in_chars <= ISOLATE_LIMIT, Error, "Source buffer is too large to transcode");
 
+  if (length_in_chars == 0) {
+    auto empty = jsg::BackingStore::alloc<v8::Uint8Array>(js, 0);
+    return jsg::BufferSource(js, kj::mv(empty));
+  }
+
   Converter from(fromEncoding);
-  auto destbuf = kj::heapArray<UChar>(length_in_chars);
+  auto destBuf = jsg::BackingStore::alloc<v8::Uint8Array>(js, length_in_chars);
+  auto destPtr = destBuf.asArrayPtr<char16_t>();
   auto actual_length =
-      simdutf::convert_latin1_to_utf16(source.asChars().begin(), source.size(), destbuf.begin());
+      simdutf::convert_latin1_to_utf16(source.asChars().begin(), source.size(), destPtr.begin());
 
   // simdutf returns 0 for invalid value.
   if (actual_length == 0) {
     return kj::none;
   }
 
-  return destbuf.first(actual_length).asBytes().attach(kj::mv(destbuf));
+  destBuf.limit(actual_length * sizeof(char16_t));
+  return jsg::BufferSource(js, kj::mv(destBuf));
 }
 
-kj::Maybe<kj::Array<kj::byte>> TranscodeFromUTF16(
-    kj::ArrayPtr<kj::byte> source, Encoding fromEncoding, Encoding toEncoding) {
+kj::Maybe<jsg::BufferSource> TranscodeFromUTF16(
+    jsg::Lock& js, kj::ArrayPtr<kj::byte> source, Encoding fromEncoding, Encoding toEncoding) {
   Converter to(toEncoding);
   auto substitute = kj::str(kj::repeat('?', to.minCharSize()));
   to.setSubstituteChars(substitute);
 
   auto utf16_input = kj::arrayPtr<char16_t>(
-      reinterpret_cast<char16_t*>(source.begin()), source.size() / sizeof(UChar));
+      reinterpret_cast<char16_t*>(source.begin()), source.size() / sizeof(char16_t));
```

Comment:
Nit: Technically this is correct, since UChar is equal to char16_t, but this function uses ICU which returns UChar. In the future, if things change, this would cause some issues (and many more).

Reply:
`UChar` is *exceedingly* unlikely to change and doing so would cause other issues if it did. We'd see test failures all over I would expect.